pkg/client/client_test.go

```diff
@@ -1,12 +1,761 @@
+		},
+		{
+			name: "additionalTrustBundle namespace not set, should fail",
+			helper: testHelper().withCustomNutanixPrismEndpointReader(
+				func() (*credentials.NutanixPrismEndpoint, error) {
+					return &credentials.NutanixPrismEndpoint{
+						Address: "cluster-endpoint",
+						Port:    9440,
+						CredentialRef: &credentials.NutanixCredentialReference{
+							Kind:      credentials.SecretKind,
+							Name:      "creds",
+							Namespace: "capx-system",
+						},
+						AdditionalTrustBundle: &credentials.NutanixTrustBundleReference{
+							Kind: credentials.NutanixTrustBundleKindConfigMap,
+							Name: "cm",
+						},
+					}, nil
+				},
+			),
+			expectProviderToBeNil: true,
+			expectedErr:           fmt.Errorf("failed to retrieve capx-namespace. Make sure %s env variable is set", capxNamespaceKey),
+		},
+		{
+			name: "reader returns an error, should fail",
+			helper: testHelper().withCustomNutanixPrismEndpointReader(
+				func() (*credentials.NutanixPrismEndpoint, error) {
+					return nil, fmt.Errorf("could not read config")
+				},
+			),
+			expectProviderToBeNil: true,
+			expectedErr:           fmt.Errorf("failed to create prism endpoint: %w", fmt.Errorf("could not read config")),
+		},
+	}
+	for _, tt := range tests {
+		tt := tt // Capture range variable.
+		t.Run(tt.name, func(t *testing.T) {
+			for k, v := range tt.envs {
+				t.Setenv(k, v)
+			}
+			provider, err := tt.helper.buildProviderFromFile()
+			if tt.expectProviderToBeNil {
+				assert.Nil(t, provider)
+			} else {
+				assert.NotNil(t, provider)
+			}
+			assert.Equal(t, tt.expectedErr, err)
+		})
+	}
+}
+
+func Test_readManagerNutanixPrismEndpointFromFile(t *testing.T) {
+	t.Parallel()
+	tests := []struct {
+		name              string
+		config            []byte
+		expectedEndpoint  *credentials.NutanixPrismEndpoint
+		expectedErrString string
+	}{
+		{
+			name:   "valid config",
+			config: []byte(validTestConfig),
+			expectedEndpoint: &credentials.NutanixPrismEndpoint{
+				Address: "cluster-endpoint",
+				Port:    9440,
+				CredentialRef: &credentials.NutanixCredentialReference{
+					Kind:      credentials.SecretKind,
+					Name:      "creds",
+					Namespace: "test",
+				},
+			},
+		},
+		{
+			name:              "invalid config, expect an error",
+			config:            []byte("{{}"),
+			expectedErrString: "failed to unmarshal config: invalid character '{' looking for beginning of object key string",
+		},
+		{
+			name:              "missing CredentialRef, expect an error",
+			config:            []byte(invalidTestConfigMissingCredentialRef),
+			expectedErrString: "credentialRef must be set on CAPX manager",
+		},
+	}
+	for _, tt := range tests {
+		tt := tt // Capture range variable.
+		t.Run(tt.name, func(t *testing.T) {
+			t.Parallel()
+			f, err := os.CreateTemp(t.TempDir(), "config-*.json")
+			assert.NoError(t, err, "error creating temp file")
```

Comment:
this should be a `require.NoError`.

Reply:
Yep good catch, test can't proceed if that fails